src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.cs

```diff
@@ -490,10 +510,12 @@ public void Dispose()
             _certContext = null;
             if (certContext != null && !certContext.IsInvalid)
             {
-                if (!_certContextCloned)
+                if (_certContextCloned)
                 {
-                    certContext.Dispose();
+                    certContext.DangerousRelease();
```

Comment:
Done; checks for null parent (and throws) but only asserts for IsInvalid and IsClosed.